### DISCUSSION TITLE
How do IF activities pass parameters in Elsa 3.1.3 designer?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5703] [General]
How do IF activities pass parameters in Elsa 3.1.3 designer?

**Comment by glime-ai** [comment]
In Elsa 3.1.3, when using the IF activity, you don't directly pass parameters to the "Then" or "Else" branches. Instead, you define a condition that evaluates to either `true` or `false`. Based on this condition, the workflow execution will continue with the activities defined in the "Then" branch if the condition is `true`, or with the activities in the "Else" branch if the condition is `false`.

To use the IF activity, you only need to specify the Condition attribute. When you click on the "Then" or "Else" buttons in the activity display, you are taken to a sub-flowchart where you can drag and drop new activities to define the logic for each branch. The parameters or data that these activities work with can be sourced from the workflow's input, variables, or other activities' outputs.

If you need to pass specific parameters to activities within the "Then" or "Else" branches, you would typically use variable expressions or input expressions that reference data available in the workflow context.

For more detailed information on how to work with activities and pass data within Elsa workflows, you can refer to the official Elsa documentation and community resources such as Stack Overflow discussions or GitHub issues where similar topics are discussed.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
